### PR TITLE
PBI 94 119 authentication page UI

### PR DESCRIPTION
**Pull request by bryannaufal**
## Describe your changes

- Implemented the Login Page component UI for authentication (not yet integrated with backend services).

## Issue ticket number and link

- ADU-94: https://linear.app/adudu/issue/ADU-94/page-ui
- ADU-119: https://linear.app/adudu/issue/ADU-119/[unit-test]-page-ui

### CODE REVIEW

**Comment by bryannaufal** [comment]
image

Login Page Preview

File: src/components/features/login-form.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react'
+
+import { Button } from '@/components/atoms/button.tsx'
+import { Input } from '@/components/atoms/input.tsx'
+import { Typography } from '@/components/atoms/typography.tsx'
+
+const LoginForm: React.FC = () => {
```
Comment (FreeJ1nG) [inline src/components/features/login-form.tsx]: Can we change this to use `function`, like we discussed previously

File: src/components/features/login-form.tsx
Comment (FreeJ1nG) [inline src/components/features/login-form.tsx:1]: This file should be in `src/components/organisms` instead of `/features`, let's reserve features for tables specifically

**Review by FreeJ1nG — CHANGES_REQUESTED** [review]

File: src/components/features/login-form.tsx (hunk 1 above)
Comment (bryannaufal) [reply src/components/features/login-form.tsx]: Thank you for reminding me, Solved

File: src/components/features/login-form.tsx
Comment (bryannaufal) [reply src/components/features/login-form.tsx:1]: Alright, I've already solved it. However, I would like to ask, how do we determine whether a component should be placed in organisms, molecules, or features?

File: src/components/organisms/login-form.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    if (errors.length > 0) {
+      setErrorMessages(errors)
+      return
+    }
+
+    setErrorMessages([])
+
+    console.log('Logging in with:', { username, password })
```
Comment (FreeJ1nG) [inline src/components/organisms/login-form.tsx]: Also, remove this `console.log()`

File: src/components/features/login-form.tsx
Comment (FreeJ1nG) [reply src/components/features/login-form.tsx:1]: https://atomicdesign.bradfrost.com/chapter-2/

We're following the principles stated there

TL;DR:
- Atoms contain smaller pieces of UI, buttons, inputs, etc
- Molecules are smaller pieces of UI that may consist of multiple atoms, 
Screenshot 2024-10-17 at 12 58 09
That example is a good one since it shows that a "searchbar" molecule consists of multiple UI pieces
- Organisms may look like something like a navigation bar


It's just a way to describe UI that makes things much more reusable

Comment (edutjie) [reply src/components/features/login-form.tsx:1]: Nice knowledge sharing Drew! Now I learn a bit about atomic design 🤩

File: src/components/atoms/input.tsx [hunk 3]
```diff
@@ -1,20 +1,40 @@
...
+        {isPassword && (
+          <button
+            type="button"
+            onClick={() => setShowPassword(!showPassword)}
+            className="absolute right-2 top-1/2 transform -translate-y-1/2 focus:outline-none"
+          >
+            {showPassword ? <EyeOff className="h-5 w-5" /> : <Eye className="h-5 w-5" />}
+          </button>
+        )}
+      </div>
```
Comment (edutjie) [inline src/components/atoms/input.tsx]: I have implemented the password hidden mechanism to the abstract component (Input) in my [PR](https://github.com/kg-procurement/kg-procurement-ui/pull/25). We can change it to that later in the integration PR because the my PR isn't merged yet.

Comment (FreeJ1nG) [reply src/components/atoms/input.tsx]: Hey, I finally merged @edutjie's PR, so please kindly merge his features and use his implementation of toggleable hidden input for password

CC: @bryannaufal

**Comment by FreeJ1nG** [comment]
LGTM, we should start removing footers from each individual page and move it to `__root.tsx`, but we can do that later

**Comment by FreeJ1nG** [comment]
Merging
